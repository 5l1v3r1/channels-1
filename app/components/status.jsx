// @flow
import React from 'react';
import { IconArrowDropDown } from './icons';
import styles from './status.scss';

const Status = ({
  thumbnail,
  title,
  toggleChannelList
}: {
  thumbnail: string,
  title: string,
  toggleChannelList: () => void
}) => (
  <div className={styles.status} onClick={toggleChannelList}>
    <img className={styles.thumbnail} src={thumbnail} />
    <h1 className={styles.title}>{title}</h1>
    <IconArrowDropDown styles={styles} />
  </div>
);

export default Status;
