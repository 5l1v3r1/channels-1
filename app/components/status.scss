@import '../app.global.scss';

.status {
  display: flex;
  align-items: center;
  height: 100%;
  padding: 0 11px;
  background-color: $red;
}

.thumbnail {
  width: 45px;
  margin-right: 11px;
  border: 2px solid $white;
  border-radius: 50%;
  box-shadow: 0 1px 3px 1px rgba(46, 46, 46, 0.3);
}

.title {
  width: 100px;
  color: $white;
  font-weight: 300;

  @include font-size(24);
}

.css-truncate-target {
  max-width: 100px !important;
}

.iconArrowDropDown {
  width: 14px;
  margin-left: 0.5rem;
  fill: $white;
}
