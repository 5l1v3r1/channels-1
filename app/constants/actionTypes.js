// @flow
const actionTypes = {
  FETCH_ACCESS_TOKEN_REQUEST: 'FETCH_ACCESS_TOKEN_REQUEST',
  FETCH_ACCESS_TOKEN_SUCCESS: 'FETCH_ACCESS_TOKEN_SUCCESS',
  FETCH_ACCESS_TOKEN_FAILURE: 'FETCH_ACCESS_TOKEN_FAILURE',
  FETCH_CHANNELS_REQUEST: 'FETCH_CHANNELS_REQUEST',
  FETCH_CHANNELS_SUCCESS: 'FETCH_CHANNELS_SUCCESS',
  FETCH_CHANNELS_FAILURE: 'FETCH_CHANNELS_FAILURE',
  UPDATE_ACTIVE_CHANNEL: 'UPDATE_ACTIVE_CHANNEL',
  FETCH_VIDEOS_REQUEST: 'FETCH_VIDEOS_REQUEST',
  FETCH_VIDEOS_SUCCESS: 'FETCH_VIDEOS_SUCCESS',
  FETCH_VIDEOS_FAILURE: 'FETCH_VIDEOS_FAILURE'
};

export default actionTypes;
