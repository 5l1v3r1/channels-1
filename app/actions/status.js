// @flow
import actionTypes from '../constants/actionTypes';
import type { Action } from '../constants/typeAliases';

export function toggleChannelList(): Action {
  return {
    type: actionTypes.TOGGLE_CHANNEL_LIST
  };
}

export function updateStatus(title: string, thumbnail: string): Action {
  return {
    type: actionTypes.UPDATE_STATUS,
    payload: {
      title,
      thumbnail
    }
  };
}
