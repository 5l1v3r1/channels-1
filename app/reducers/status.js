// @flow
import actionTypes from '../constants/actionTypes';
import type { Action, StatusState } from '../constants/typeAliases';
import placeholder from '../images/placeholder.jpg';

const initialState = {
  thumbnail: placeholder,
  title: 'Channels'
};

export default function status(state: StatusState = initialState, { type, payload }: Action): StatusState {
  switch (type) {
    case actionTypes.UPDATE_STATUS:
      return {
        title: payload.title,
        thumbnail: payload.thumbnail
      };
    default:
      return state;
  }
}
