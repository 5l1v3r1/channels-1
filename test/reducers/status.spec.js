import status from '../../app/reducers/status';
import actionTypes from '../../app/constants/actionTypes';

describe('status reducer', () => {
  it('should return the initial state', () => {
    expect(status(undefined, {})).toMatchSnapshot();
  });

  it('should handle UPDATE_STATUS', () => {
    const action = {
      type: actionTypes.UPDATE_STATUS,
      payload: {
        thumbnail: 'https://yt3.ggpht.com/photo.jpg',
        title: 'DevTips'
      }
    };
    expect(status(undefined, action)).toMatchSnapshot();
  });

  it('should handle unknown action type', () => {
    const action = { type: 'unknown' };
    expect(status(undefined, action)).toMatchSnapshot();
  });
});
