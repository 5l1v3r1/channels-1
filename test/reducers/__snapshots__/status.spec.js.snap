// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`status reducer should handle UPDATE_STATUS 1`] = `
Object {
  "thumbnail": "https://yt3.ggpht.com/photo.jpg",
  "title": "DevTips",
}
`;

exports[`status reducer should handle unknown action type 1`] = `
Object {
  "thumbnail": "test-file-stub",
  "title": "Channels",
}
`;

exports[`status reducer should return the initial state 1`] = `
Object {
  "thumbnail": "test-file-stub",
  "title": "Channels",
}
`;
